deploy.sh

```diff
@@ -13,55 +13,52 @@ help() {
 EOH
 }
 
-
 check_environment() {
     # verify that the proper environment is passed
     local environment="${1?No environment passed}"
 
     if [[ ! -d "deploy/$environment" || "$environment" =~ ^(base|README)$ ]]; then
         echo "Unknown environment $environment, use one of:"
-        ls deploy/ | egrep -v '^(base|README)'
+        find deploy/ | grep -E '^(base|README)'
```

Comment:
Thanks for working on this @blancadesal. I noticed that substituting "find" for "ls" here doesn't solve the issue at all. Executing this prints nothing to the commandline at all.

You also removed "-v" arg, but even with "-v" (provided we are still using "find"), the result is still undesirable. As an example, executing `find deploy/ | grep -vE '^(base|README)'` gives us this:

`
deploy/
deploy/tools
deploy/tools/kustomization.yaml
deploy/tools/auth-patch.yaml
deploy/base
deploy/base/tekton-controller-rbac.yaml
deploy/base/kustomization.yaml
deploy/base/pipeline.yaml
deploy/base/namespace.yaml
deploy/base/toolforge-buildpacks-phases-patch.json
deploy/base/rbac.yaml
deploy/base/auth.yaml
.................
`

I don't have the answer to this problem though. When I encountered it I just told shellcheck to ignore that line.

Reply:
`find deploy/ -mindepth 1 -maxdepth 1 -type d ! -name 'base' ! -name 'README' -exec basename {} \; | grep -Ev '^(base|README)' ` produces the intended output, at a cost to readability.

The rationale given for the `ls` --> `find` switch is that...

> `ls` is only intended for human consumption: it has a loose, non-standard format and may "clean up" filenames to make output easier to read.
https://www.shellcheck.net/wiki/SC2012

In other words, `find` handles directory- and filenames containing non-alphanumeric characters better. This is quite a rare edge case, though. I'd favor just skipping this rule altogether. Thoughts? @NdibeRaymond

Reply:
Slightly shorter version:

```
find deploy -maxdepth 1 -mindepth 1 -type d ! -regex 'deploy/base.*' -exec basename {} \;
```

Reply:
@dhinus what are your thoughts on this lint rule in general? If we start using the shellcheck linter here, we will eventually add it to other repos containing shell scripts as well.

Reply:
I have not used shellcheck before, but I think the suggestion of using `find` here is reasonable... I don't have an opinion on shellcheck in general because I haven't looked at all the rules, but I think it's worth a try!

Reply:
In my experience shellcheck is useful, specially if you are not very familiar with bash, as there's many unexpected behaviors that shellcheck tries to avoid.

Said that, sometimes it's not worth the trouble trying to adapt to all the edge cases if you know that they will not happen, or would be easy to catch (as it can get complicated to catch all the issues, see https://www.youtube.com/watch?v=-5wpm-gesOY xd).

Given that we manage the directories inside that one, that it will always be a short list of well-named ones, I think the suggestion is not needed (but I'm ok if you want to follow it too).

Reply:
I did change it to @dhinus version above. It's a bit long but, I think, not hideous enough to disable the rule altogether, for now.

Reply:
LGTM, but I'm having issues with pre-commit... will merge once I sort it out :/